### PR TITLE
[#185] Update to ligo 0.46.1

### LINKED ISSUE #185: Bump LIGO version and mention the latest tested one
# Clarification and motivation

It seems that the latest LIGO version used by CI to compile the contract is [0.2.1](https://gitlab.com/ligolang/ligo/-/releases/0.2.1).

Moreover currently in the `ligo/README.md`'s "Build Instructions" documentation we say that:
> Since LIGO is an actively developed project, some versions of it may be incompatible with Stablecoin.
The revision specified in the sources.json file is guaranteed to be compatible.

which is true, but isn't very clear or easy to figure out.

# Acceptance criteria

1. The LIGO version is bumped to a more recent [released one](https://gitlab.com/ligolang/ligo/-/releases) (if possible, the latest available)
2. The documentation mentions the latest tested LIGO version's number instead of redirecting to the source code

### PR DESCRIPTION
**Pull request by dcastro**
## Description

We're using a really old version of LIGO, 0.10.0. This PR updates the contract to be compatible with the latest version, 0.46.1

## Related issue(s)

Resolves #185 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock
  - [x] I updated [changelog](../tree/master/ChangeLog.md) unless I am sure my changes are
        not essential.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: ligo/stablecoin/actions.ligo [hunk 1]
```diff
@@ -39,13 +39,13 @@
 function authorize_pending_owner
   ( const store : storage
   ; const full_param : closed_parameter
-  ) : storage * address is case store.roles.pending_owner of
-    Some (pending_owner) ->
+  ) : storage * address is case store.roles.pending_owner of [
```
Comment (sras) [inline ligo/stablecoin/actions.ligo:42]: Is the `[  ]` around  patterns a new addition to LIGO ?

Comment (dcastro) [reply ligo/stablecoin/actions.ligo:42]: Yup, it's required now, it replaces the `end` keyword.

File: ligo/stablecoin/actions.ligo [hunk 2]
```diff
@@ -103,44 +103,44 @@ function ensure_is_paused
...
+  | Some (ledger_balance) -> updated_balance := ledger_balance
   | None -> skip
-  end
+  ]
 ; updated_balance := updated_balance + parameter.amount
 ; const updated_ledger : ledger =
     Big_map.update(parameter.to_, Some (updated_balance), store.ledger)
 } with store with record [ ledger = updated_ledger ]
 
-type debit_param_ is record
```
Comment (sras) [inline ligo/stablecoin/actions.ligo:119]: I can't remember why we had this "_" suffix for these types, but is this no longer required?

Comment (sras) [reply ligo/stablecoin/actions.ligo:119]: Oh, I remember now. So these labels are not forbidden anymore, right?

**Review by sras — APPROVED** [review]
LGTM

**Pull request merged**